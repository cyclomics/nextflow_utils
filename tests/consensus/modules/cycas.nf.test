nextflow_process {

    name "Test Process Cycas"
    script "consensus/modules/cycas.nf"
    process "Cycas"
    
    
    test("Cycas consensus") {

        when {
            process {
                """
                sample_id = "test_sample"
                bam_file = file("$baseDir/example_datasets/example_native_reads/bam/FAW79986_pass_dbcafcd9_40623079_0_filtered.NM_50_mapq_20.bam")
                bai_file = file("$baseDir/example_datasets/example_native_reads/bam/FAW79986_pass_dbcafcd9_40623079_0_filtered.NM_50_mapq_20.bam.bai")

                input[0] = Channel.of([sample_id, bam_file, bai_file])
                """
            }
        }
        then {
            assert process.success
            // assert snapshot(process.out).match() // failing idk why
            
            with(process.out[0]) {
                assert path(get(0)[1]).readLines().size() == 11493
            }
        }
    }
}
