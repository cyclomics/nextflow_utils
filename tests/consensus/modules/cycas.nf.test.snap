{
    "Cycas consensus": {
        "content": [
            {
                "0": [
                    [
                        "test_sample",
                        "FAW79986_pass_dbcafcd9_40623079_0_filtered.consensus.fastq:md5,a4fc82a8af79cbceb83d7ea99e7ae48f",
                        "FAW79986_pass_dbcafcd9_40623079_0_filtered.metadata.json:md5,9a1f2dfc99a0f6bb4a6b2f11dd6cec5c"
                    ]
                ]
            }
        ],
        "timestamp": "2024-01-10T17:12:17.370452"
    }
}